rust/geoarrow-geo/src/contains.rs

```diff
@@ -0,0 +1,132 @@
+use arrow_array::BooleanArray;
+use geo::contains::Contains;
+use geo_traits::to_geo::ToGeoGeometry;
+use geoarrow_array::GeoArrowArrayAccessor;
+use geoarrow_schema::error::GeoArrowResult;
+
+pub fn contains<'a>(
+    left_array: &'a impl GeoArrowArrayAccessor<'a>,
+    right_array: &'a impl GeoArrowArrayAccessor<'a>,
+) -> GeoArrowResult<BooleanArray> {
```

Comment:
Similarly to `intersects`, we want the public API to be in terms of `&dyn GeoArrowArray`, not `GeoArrowArrayAccessor`. The latter would require all users to manually downcast their data to a concrete type, which we don't want to require them to do.

Follow the layout of `intersects` which has a separate inner function that takes in `GeoArrowArrayAccessor`